// react-material-music-player/dist/components/Playlist/PlaylistItemTemplate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _Menu2 = _interopRequireDefault(require("@mui/icons-material/Menu"));
var _PlayArrowRounded2 = _interopRequireDefault(require("@mui/icons-material/PlayArrowRounded"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Box = _interopRequireDefault(require("@mui/material/Box"));
var _CoverArt = _interopRequireDefault(require("../CoverArt"));
var _withoutPropagation = _interopRequireDefault(require("../../utils/withoutPropagation"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
class PlaylistItemTemplate extends _react.default.Component {
  constructor(props) {
    super(props);
    this.props = props;
  }
  handleSelect() {
    const commonProps = this.props.commonProps;
    const item = this.props.item;
    commonProps.onTrackSelect(commonProps.listOfID.indexOf(item.ID));
  }
  render() {
    const {
      item,
      dragHandleProps,
      commonProps,
      itemSelected
    } = this.props;
    return /*#__PURE__*/_react.default.createElement(_Box.default, {
      sx: {
        maxHeight: "100%",
        // requirement for best results
        // flex box
        display: "flex",
        flexDirection: "row",
        alignItems: "center",
        flexWrap: "nowrap",
        // border
        border: 1,
        borderColor: "divider",
        borderRadius: 1,
        // paddding
        boxSizing: "border-box",
        padding: 0.5,
        // elvate when selected
        boxShadow: itemSelected > 0 ? 16 : 0
      }
    }, /*#__PURE__*/_react.default.createElement(_Box.default, {
      sx: {
        display: "flex",
        flexGrow: 1,
        alignItems: "center"
      },
      onClick: (0, _withoutPropagation.default)(this.handleSelect.bind(this))
    }, commonProps.currentTrackID === item.ID ? /*#__PURE__*/_react.default.createElement(_PlayArrowRounded2.default, null) : /*#__PURE__*/_react.default.createElement(_Box.default, {
      sx: {
        width: "24px",
        height: "24px"
      }
    }), /*#__PURE__*/_react.default.createElement(_CoverArt.default, {
      src: item.coverArt,
      sx: {
        height: "48px",
        width: "48px"
      }
    }), /*#__PURE__*/_react.default.createElement(_Box.default, {
      sx: {
        width: "50px",
        flexGrow: 1,
        flexShrink: 1,
        mx: 1
      }
    }, /*#__PURE__*/_react.default.createElement(_Box.default, {
      sx: {
        width: "100%",
        typography: "subtitl3",
        whiteSpace: "nowrap",
        overflow: "hidden"
      }
    }, item.title), /*#__PURE__*/_react.default.createElement(_Box.default, {
      sx: {
        width: "100%",
        typography: "subtitle2",
        whiteSpace: "nowrap",
        overflow: "hidden"
      }
    }, item.artist))), /*#__PURE__*/_react.default.createElement(_Menu2.default, _extends({
      sx: {
        mx: 1
      }
    }, dragHandleProps, {
      onClick: e => {
        e.stopPropagation();
      }
    })));
  }
}
PlaylistItemTemplate.propTypes = {
  classes: _propTypes.default.object.isRequired
};
var _default = exports.default = PlaylistItemTemplate;
