// react-text-to-speech/dist/utils.js
import { isValidElement } from "react";
export function ArrayToText(element) {
    if (typeof element === "string")
        return element;
    return element.map(ArrayToText).join(" ") + " ";
}
export function JSXToArray(element) {
    if (isValidElement(element)) {
        const { children } = element.props;
        if (Array.isArray(children))
            return children.map(JSXToArray);
        return JSXToArray(children);
    }
    return typeof element === "string" ? element : typeof element === "number" ? String(element) : "";
}
export function cancel() {
    var _a;
    if (typeof window !== "undefined")
        (_a = window.speechSynthesis) === null || _a === void 0 ? void 0 : _a.cancel();
}
export function TextToChunks(text, desktopSize = 1000, mobileSize = 250) {
    const length = text.length;
    const size = isMobile() ? mobileSize : desktopSize;
    const result = [];
    let startIndex = 0;
    while (startIndex < length) {
        let endIndex = Math.min(startIndex + size, length);
        if (endIndex < length && text[endIndex] !== " ") {
            const spaceIndex = text.lastIndexOf(" ", endIndex);
            if (spaceIndex > startIndex)
                endIndex = spaceIndex;
        }
        result.push(text.slice(startIndex, endIndex));
        startIndex = endIndex;
    }
    return result;
}
export function findCharIndex(words, index) {
    let currentIndex = 0;
    function recursiveSearch(stringArray, parentIndex = "") {
        if (typeof stringArray === "string") {
            const elementIndex = index - currentIndex;
            return (currentIndex += stringArray.length + 1) > index ? getIndex(parentIndex, elementIndex) : "";
        }
        for (let i = 0; i < stringArray.length; i++) {
            const element = stringArray[i];
            const result = recursiveSearch(element, i);
            if (result)
                return getIndex(parentIndex, result);
        }
        currentIndex++;
        return "";
    }
    return recursiveSearch(words);
}
export const getIndex = (parentIndex, index) => `${parentIndex === "" ? "" : parentIndex + "-"}${index}`;
export function isMobile(iOS = true) {
    var _a;
    let result = (_a = navigator.userAgentData) === null || _a === void 0 ? void 0 : _a.mobile;
    result !== null && result !== void 0 ? result : (result = /Android|webOS|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent) || (iOS && /iPhone|iPad|iPod/i.test(navigator.userAgent)));
    return result;
}
export function isParent(parentIndex, index) {
    if (!(index === null || index === void 0 ? void 0 : index.startsWith(parentIndex)))
        return false;
    if (parentIndex) {
        const indexParts = index.split("-");
        const parentIndexParts = parentIndex.split("-");
        for (let i = 0; i < parentIndexParts.length; i++) {
            if (indexParts[i] !== parentIndexParts[i])
                return false;
        }
    }
    return true;
}
export const sanitize = (text) => text.replace(/[<>]|(&[^\s;]+);/g, (match, group) => (match === "<" ? "(" : (group || "") + ")"));
