// react-material-music-player/dist/components/TrackDetials.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = TrackDetails;
var _react = _interopRequireDefault(require("react"));
var _reactRedux = require("react-redux");
var _Box = _interopRequireDefault(require("@mui/material/Box"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function TrackDetails(props) {
  const {
    title,
    artist
  } = (0, _reactRedux.useSelector)(/**@type {import("../redux/types").useSelectCb} */
  state => {
    let currentTrack = state.playlist[state.currentTrack];
    return {
      title: currentTrack?.title,
      artist: currentTrack?.artist
    };
  }, _reactRedux.shallowEqual);
  const sx = props.sx;
  return /*#__PURE__*/_react.default.createElement(_Box.default, {
    sx: sx
  }, /*#__PURE__*/_react.default.createElement(_Box.default, {
    sx: {
      typography: "subtitl3",
      whiteSpace: "nowrap",
      overflow: "hidden",
      textOverflow: "ellipsis",
      width: "100%"
    }
  }, title), /*#__PURE__*/_react.default.createElement(_Box.default, {
    sx: {
      typography: "subtitle2",
      whiteSpace: "nowrap",
      overflow: "hidden",
      textOverflow: "ellipsis",
      width: "100%"
    }
  }, artist));
}
